postprocessing/science/compute_isochrones.py

```diff
@@ -0,0 +1,250 @@
+    
+    threshold = np.amax(absWaveform)*trigger           # trigger dictates the portion of the maximum ground velocity,
+    t = np.argmax(absWaveform>threshold)*dtSurface     # which needs to be reached to pick the p-phase
+    return t                                           # p-phase arrival time (equal to traveltime if nucleation at t=0)
+
+
+def ApproximateHypocenter(absoluteSliprates, faultCells, slipRateThreshold=args.sliprateThreshold[0]):
+    
+    slippingFault = np.where(absoluteSliprates > slipRateThreshold, 1, 0)
+    slippingFault = np.where(slippingFault > 0, slippingFault - slippingFault[:,0].reshape(-1,1), 0) 
+    # ignore elements that already slip at t=0 (important when the output contains more than one event)
+    
+    indHypocenter = np.argwhere(slippingFault[:,np.argmax(np.sum(slippingFault, axis=0) > 100)]==1)
+    hypocenter = np.mean(faultCells[indHypocenter,:], axis=0)[0]
+    return hypocenter
+
+
+def GetTraveltimesToAllFaultCells(receiverCoords, faultCells, swaveVelocity):
+    
+    distances = np.sqrt(np.sum((receiverCoords - faultCells) ** 2, axis=1))
+    return distances * (1 / swaveVelocity)
+
+
+def MedianSmoothing(input_tuple):
+    # input_tuple: (parameter, xyz, indices, radius)
+    parameter = input_tuple[0]
+    xyz = input_tuple[1]
+    indices = input_tuple[2]
+    radius = input_tuple[3]
+    parameter_new = np.zeros(indices.shape[0])
+    for i, j in enumerate(indices):
+        parameter_new[i] = np.median(parameter[(xyz[:,0] < xyz[j,0]+radius) & (xyz[:,0] > xyz[j,0]-radius) & 
+                            (xyz[:,1] < xyz[j,1]+radius) & (xyz[:,1] > xyz[j,1]-radius) &                    
+                            (xyz[:,2] < xyz[j,2]+radius) & (xyz[:,2] > xyz[j,2]-radius)])
+    return parameter_new
+
+
+def MedianSmoothingParallel(parameter, xyz, radius=250., nprocs=4):
+    
+    chunks = [tuple((parameter, xyz, i, radius)) for i in np.array_split(np.arange(parameter.shape[0]), nprocs)]
+        
+    p = Pool(nprocs)
+    result = p.map(MedianSmoothing, chunks)
+    p.close()
+    p.join()
+    parameter_new = np.concatenate(result)
+    return parameter_new
+
+
+start = timeit.default_timer()
+
+print("Loading data...")
+surfaceXdmf = seissolxdmf.seissolxdmf(args.filename)
+surfaceCells = ComputeTriangleMidpoints(surfaceXdmf.ReadGeometry(), surfaceXdmf.ReadConnect())
+dtSurface = surfaceXdmf.ReadTimeStep()
+
+if not args.faultXdmf:
+    args.faultXdmf = args.filename[:-12]+"fault.xdmf"
+    
+faultXdmf = seissolxdmf.seissolxdmf(args.faultXdmf) 
+faultGeom = faultXdmf.ReadGeometry()
+faultConnect = faultXdmf.ReadConnect()
+faultCells = ComputeTriangleMidpoints(faultGeom, faultConnect)
+dtFault = faultXdmf.ReadTimeStep()
+timeIndicesFault = GetTimeIndices(faultXdmf)
+ruptureTimes = faultXdmf.ReadData("RT", timeIndicesFault[1]-1)
+
+if args.events[0] == 2:
+    ruptureTimes = ruptureTimes - faultXdmf.ReadData("RT", timeIndicesFault[0]-1)
```

Comment:
`ruptureTimes = ruptureTimes - faultXdmf.ReadData("RT", timeIndicesFault[0])`
?

Reply:
this sets all RTs to zero that slipped before timeIndicesFault[0]